sdk/identity/identity/src/credentials/interactiveBrowserCredential.ts

```diff
@@ -4,32 +4,30 @@
 /* eslint-disable @typescript-eslint/no-unused-vars */
 
 import { TokenCredential, GetTokenOptions, AccessToken } from "@azure/core-http";
-import { InteractiveBrowserCredentialOptions } from "./interactiveBrowserCredentialOptions";
+import {
+  InteractiveBrowserCredentialOptions,
+  AuthenticationRecord
+} from "./interactiveBrowserCredentialOptions";
 import { credentialLogger, formatError } from "../util/logging";
 import { TokenCredentialOptions, IdentityClient } from "../client/identityClient";
 import { DefaultTenantId, DeveloperSignOnClientId } from "../constants";
 import { Socket } from "net";
 
-const SERVER_PORT = process.env.PORT || 80;
-
 import express from "express";
-import { PublicClientApplication, TokenCache, AuthorizationCodeRequest } from "@azure/msal-node";
+import {
+  PublicClientApplication,
+  TokenCache,
+  AuthorizationCodeRequest,
+  Configuration
+} from "@azure/msal-node";
 import open from "open";
 import path from "path";
 import http from "http";
+import { CredentialUnavailable } from "../client/errors";
 
-const BrowserNotSupportedError = new Error(
-  "InteractiveBrowserCredential is not supported in Node.js."
-);
 const logger = credentialLogger("InteractiveBrowserCredential");
 
-interface AuthenticationRecord {
-  authority?: string;
-  homeAccountId: string;
-  environment: string;
-  tenantId: string;
-  username: string;
-}
+class AuthenticationRequired extends CredentialUnavailable {}
```

Comment:
might be worth setting a custom name on the error class. It also feels a little bad to not have the suffix `Error` here, as **all** the built-in JS errors are suffixed with `Error`.

Oh and if you're going to make a custom error type, shouldn't you export it somewhere?

Reply:
You can see how we set the prototype and name in RestError here: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/restError.ts